Devops deployment 6 create and test noctua developer deployment docs

mugitty opened the pull request:
For geneontology/devops-documentation#6

mugitty committed:
For geneontology/devops-documentation/#6

mugitty committed:
For devops-development/#6

mugitty committed:
For devops-deployment/#6

mugitty committed:
For devops-development/#6

mugitty committed:
For geneontology-devops-deployment/#6

mugitty committed:
For devops-deployment#6

mugitty committed:
For devops-deployment#6

mugitty committed:
For geneontology/noctua_app_stack#10

mugitty committed:
Merge branch 'main' into devops-deployment-6-create-and-test-noctua-developer-deployment-docs

mugitty committed:
For geneontology/devops-documentation#6

mugitty committed:
Merge branch 'main' into devops-deployment-6-create-and-test-noctua-developer-deployment-docs

mugitty committed:
Merge branch 'main' into devops-deployment-6-create-and-test-noctua-developer-deployment-docs

mugitty committed:
For devops-development#6

mugitty committed:
For geneontology/noctua_app_stack#10

mugitty committed:
For geneontology/devops-documentation#6

mugitty committed:
For geneontology/devops-documentation#6

kltm reviewed (commented):
@mugitty Overall this looks good--thank you!

I was wondering about the removal of the bot filtering from the apache vhost setups. As we've bot issues in the past, we put those in. Was there a problem with them?

kltm reviewed (commented):
Would it be possible to remove or elide hyper specific information, like ssh fingerprint information?

kltm commented on production/PRODUCTION_README.md:
```diff
@@ -1,10 +1,289 @@
+####  Copy the ssh keys from your docker host into the running docker image, in `/tmp`:
+
+These commands may have to be executed from a separate terminal that can run Docker commands.
+See Prerequisites 2 for keys and copy keys from SpderOak
```
Could you remove the reference to SpiderOak here? We'll do that for developer onboarding (hopefully). I'd ideally like to avoid some specifics here. Paranoia.

mugitty replied:
Agreed - Security is paramount.  I have updated the documentation to remove reference to SpiderOak

kltm commented:
This is really looking good!

mugitty commented:
@kltm, I will update as requested and let you know when it is ready

mugitty committed:
For geneontology/devops-documentation#6

mugitty commented:
> @mugitty Overall this looks good--thank you!
> 
> I was wondering about the removal of the bot filtering from the apache vhost setups. As we've bot issues in the past, we put those in. Was there a problem with them?

Thanks for flagging.  Rewrite rule has precedence over Redirect and order in configuration file does not make a difference.  I updated the Apache vhost configuration files to include RewriteRule for bots

mugitty commented:
> @kltm, I will update as requested and let you know when it is ready

I have addressed the issues @kltm  mentioned.  I also made a minor update to the documentation about checking out Minerva code before building.

kltm commented:
@mugitty Great--thank you! I'll check again.

kltm committed:
remove trailing whitespace

kltm committed:
some formatting standardizing

kltm committed:
copyediting

kltm committed:
typo

kltm committed:
some corrections around neo vs golr; for geneontology/pipeline#407

kltm committed:
fox some documentation and adjust example file; for geneontology/devops-documentation#6

kltm committed:
clarify core software updates and how to change workbenches; for #14

kltm committed:
correct how to change workbenches; for #14

kltm committed:
update workbenches for next build; for #14

kltm committed:
small changes as I do another walkthrough; for #14

kltm committed:
doc clarification and (possibly) fixing checkout for workbench; for #14

kltm committed:
doc clarification and (possibly) fixing checkout for workbench; for #14

kltm committed:
something simpler for devops docs, updating templates as well, trying to get this more turnkey; for geneontology/devops-documentation#6

kltm committed:
quick finish on simplification pass; for geneontology/devops-documentation#6

kltm committed:
small tuning as I run through docs again; for geneontology/devops-documentation#6

kltm committed:
github oauth list; for geneontology/devops-documentation#6

kltm committed:
correct error in YAML during refactor/cleaning; for geneontology/devops-documentation#6

kltm committed:
turns out production-ification was easier than I thought; for geneontology/devops-documentation#6

kltm committed:
clarify reattaching; for geneontology/devops-documentation#6

kltm committed:
small add; better default golrs (work); for geneontology/devops-documentation#6

kltm committed:
updates to oauth2, blazegraph, amigo; geneontology/devops-documentation#6

kltm committed:
add production instruction for aws s3 model saving; start github and SOP thoughts; for #7 and geneontology/developer-devops#6

kltm committed:
first instruction draft for non-standard noctua-models location; for geneontology/noctua_app_stack#7

kltm committed:
make things easier to see; for geneontology/noctua_app_stack#7

kltm committed:
small fixes; for geneontology/noctua_app_stack#7

kltm committed:
add warning; for geneontology/noctua_app_stack#7

kltm committed:
Clarify GitHub OAuth credentials instructions

Clarified instructions for obtaining GitHub OAuth credentials and added notes on defining them later in the README.

kltm committed:
Update README with server usage clarification

Clarify usage of available servers for testing.

kltm committed:
Modify README for production deployment commands

Updated README to include production file commands.

kltm committed:
remove get_md5 for geneontology/noctua_app_stack#16
